docs/logs/getting-started-console-di/Program.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry;
+using OpenTelemetry.Logs;
+
+// Add services to the container, including OpenTelemetry with logging
+var services = new ServiceCollection();
+services.AddOpenTelemetry().WithLogging(logging => logging.AddConsoleExporter());
+services.AddSingleton<ExampleService>();
+IServiceProvider serviceProvider = services.BuildServiceProvider();
+
+// Get the ExampleService object from the container
+ExampleService service = serviceProvider.GetRequiredService<ExampleService>();
+
+service.DoSomeWork();
+
+// This will flush the remaining logs.
+serviceProvider.GetRequiredService<LoggerProvider>().ForceFlush();
+
+// Dispose SDK before the application ends.
+serviceProvider.GetRequiredService<OpenTelemetrySdk>().Dispose();
```

Comment:
Is this the appropriate object to dispose() when using DI in console?

Reply:
Not really. What you want to do is dispose the service provider itself:

```diff
-IServiceProvider serviceProvider = services.BuildServiceProvider();
+ServiceProvider serviceProvider = services.BuildServiceProvider();
```

`BuildServiceProvider` returns concrete `ServiceProvider` which can be disposed.

```diff
// Dispose SDK before the application ends.
-serviceProvider.GetRequiredService<OpenTelemetrySdk>().Dispose();
+serviceProvider.Dispose();
```